content/hazelcast.tex

```diff
@@ -0,0 +1,175 @@
+\subsection*{Members and Clients}
+Besides the console application Hazelcast also provides the opportunity to include the Hazelcast package in your code. For Java there is a package for members and one for clients provided. Other programming languages only have clients to work with. The difference between clients and members is that clients do not hold data. They connect to Hazelcast cluster members and access the data on that way for read and write operations \cite{hazelcastmanual}. If Hazelcast has to be set up for an application in C++ for example, the scripts in the \texttt{demo/} directory can be used to start a cluster member without needing to create a java application. The C++ application than has to include the Hazelcast package and can access or create the data via the member.
+\subsection*{Configuration}
+Hazelcast offers an XML file to adjust certain configurations in the \texttt{bin/} directory. \\
+It is possible to change the amount of backups and also the types of backups. There are normal backups which are synchronized and therefore lock the data in case it is manipulated. All other nodes have to wait until the data is changed in all backups and the changes are confirmed by the nodes which hold the backups. Additionally, there are asynchronous backups. They do not lock any data and therefore bring a performance increase because the nodes do not have to wait for them to confirm the changes. On the other hand, it brings the risk of inconsistent data. Nodes could access data which are no longer valid because the changes were not made on all backups after a manipulation took place. \\
+In general, increasing the number of backups will increase the read performance because the data can be read on different nodes in parallel. The costs for this advantage are either bad write performance in case of normal backups or inconsistency in case of asynchronous backups.\\ 
+Furthermore Hazelcast provides configurations about how big a particular data structure can grow and how to act when there is no more space left. 
+\begin{flushleft}
+\begin{figure}[ht]
+    \includegraphics{content/images/hazelcastXML.PNG}
+    \caption{Hazelcast Configuration \cite{johns2015}}
+\end{figure}
+\end{flushleft}
+In this example the map called \textit{"capitals"} has a maximum size of 10 items per node. When reaching this maximum, 20 percent of the data will be freed according to the principle of Least Frequently Used (LFU). One synchronous and one asynchronous backup will be created and the time to live for data sets is set. This means that data will be deleted after this amount of seconds goes by. In contrast to this, setting the maximum idle time will only delete a data set when it is not accessed after a certain amount of seconds.\\
+\citeauthor{johns2015} and the Hazelcast websites provide more information about configuration and go more in detail.
+
+\section{Cheat Sheet}
+\subsection*{General commands}
+\begin{tabular}{|p{4cm}|p{9cm}|}
```

Comment:
Please prefer `\textwidth` over hardcoded sizes when there is no good reason not to do it.

Reply:
we prefer fix sizes

Reply:
This is bad practice. Your document won't work when the layout is changed. This could happen in other PRs, if there is a good reason for this.
Therefore, relying on the document / text width - multiplied by a factor - is preferred.

Reply:
Again, not resolved.